flair/models/lemmatizer_model.py

```diff
@@ -0,0 +1,849 @@
+                 use_attention: bool = True
+                 ):
+        """
+        Initializes a Lemmatizer model
+        The model consists of a decoder and an encoder. The decoder is a Token-Embedding from flair. The embedding of a token
+        is handed as the initial hidden state to the decoder, which is a RNN-cell (torch.nn.GRU) that predicts the lemma of
+        the given token one letter at a time.
+        Note that for learning one can use data in which only those words are annotated that differ from their lemma or data in which
+        all words are annotated with a (maybe equal) lemma.
+        :param token_embedding: Embedding used to encode sentence
+        :param input_size: Input size of the RNN's. Each letter of a token is represented by a hot-one-vector over the given character
+            dictionary. This vector is transformed to a input_size vector with a linear layer.
+        :param hidden_size: size of the hidden state of the RNN. The initial embedding is transformed to a vector of size hidden_size
+            with an additional linear layer
+        :param num_layers_in_rnn: Number of stacked RNN cells
+        :param path_to_char_dict: Path to character dictionary. If None, a standard dictionary will be used.
+            Note that there are some rules to the dictionary. The first three indices must be reserved for the special characters
+            <>, <S>, <E>, in that order. The characters represent a dummy symbol, start and end of a word, respectively.
+        :param label_type: Name of the gold labels to use.
+        :param beam_size: Number of hypothesis used when decoding the output of the RNN. Only used in prediction.
+        :param max_sequence_length_dependend_on_input: If set to True, the maximum length of a decoded sequence in the prediction
+            depends on the sentences you want to lemmatize. To be precise the maximum length is computed as the length of the longest
+            token in the sentences plus one.
+        :param max_sequence_length: If set to True and max_sequence_length_dependend_on_input is False a fixed maximum length for
+            the decoding will be used for all sentences.
+        """
+
+        super().__init__() # TODO: LSTMs instead of GRUs, maybe as a parameter?
+
+        self._label_type = label_type
+        self.beam_size = beam_size
+        self.max_sequence_length = max_sequence_length
+        self.dependend_on_input = max_sequence_length_dependend_on_input
+        self.use_attention = use_attention
+
+
+        # encoder: embed tokens and map embedding to hidden size of RNN
+        self.token_embedding = token_embedding
+        self.hidden_size = hidden_size
+        if self.token_embedding:
+            self.emb_to_hidden = nn.Linear(token_embedding.embedding_length, hidden_size)
+
+        # decoder: dictionary of characters to create one-hot embeddings for decoder inputs
+        self.path_to_char_dict = path_to_char_dict
```

Comment:
Why is this a class variable? Isn't it enough to load the dictionary at this point?

Reply:
Done